compiler/hash-source/src/location.rs

```diff
@@ -94,3 +94,14 @@ pub struct SourceLocation {
     /// The id of the source that the span is referencing.
     pub id: SourceId,
 }
+
+impl SourceLocation {
+    /// Join the span of a [SourceLocation] with another [SourceLocation].
+    ///
+    /// *Note*: the `id` of both [SourceLocation]s must be the same.
+    pub fn join(self, other: Self) -> Self {
+        assert!(self.id == other.id);
```

Comment:
Why ?

Reply:
How else can you join the spans if they have different source-ids?

Reply:
Oh right, i was just unclear on what the ID was for. So it is SourceId ? Maybe that field should be renamed..

Reply:
It was previously named `source_id` but I thought that was too verbose, I guess if it does create confusion then maybe we can revert